device/store.go

```diff
@@ -112,17 +136,40 @@ func (d *Device) isOutdated(interfaceName string, interfaceMajor int) bool {
 		}
 	}
 	// if the interface is not present in the current introspection, it must be outdated
-	return false
+	return true
 }
 
 func (d *Device) storeProperty(interfaceName string, path string, interfaceMajor int, value []byte) {
+	if d.db == nil {
+		// Nothing to do
+		return
+	}
 	d.db.Clauses(clause.OnConflict{
 		UpdateAll: true,
 	}).Create(&property{InterfaceName: interfaceName, Path: path, InterfaceMajor: interfaceMajor, RawValue: value})
 }
 
-func (d *Device) deleteProperty(interfaceName string, path string, interfaceMajor int) {
-	d.db.Delete(&interfaceName, &path, &interfaceMajor)
+func (d *Device) removePropertyFromStorage(interfaceName, path string, interfaceMajor int) {
```

Comment:
Keep naming uniform: if storing is `storeProperty`, removing should be `removeProperty` (or add `FromStorage` everywhere, as you like)

Reply:
given this is still a private method to Device, I'd keep it more specific. I'll think about something.

Reply:
Thinking about this twice - I think current semantics are fine, as `storeProperty` implies storage, whereas `deleteProperty` is super ambiguous. We could resort to `addPropertyToStorage` but in general, I'd assume wherever `store` is part of the method name it is implied storage is concerned, otherwise `FromStorage` or `ToStorage` should be added for clarity. All of this keeping in mind that all these methods are private APIs to device, so being ambiguous will eventually backfire. Let me know what you think